python/pyspark/sql/dataframe.py

```diff
@@ -2046,6 +2046,40 @@ def toDF(self, *cols):
         jdf = self._jdf.toDF(self._jseq(cols))
         return DataFrame(jdf, self.sql_ctx)
 
+    @since(3.0)
+    def transform(self, func):
+        """Returns a new class:`DataFrame` according to a user-defined custom transform method.
+        This allows chaining transformations rather than using nested or temporary variables.
```

Comment:
I would just match the doc to Scala API side as well.

Reply:
what do you mean? 
I don't see a scala documentation

Reply:
I meant Scala side `transform`'s documentation here:

https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L2485-L2492